state/core_access.go

```diff
@@ -153,7 +153,7 @@ func (ca *CoreAccessor) SubmitPayForData(
 ) (*TxResponse, error) {
 	response, err := payment.SubmitPayForData(ctx, ca.signer, ca.coreConn, nID, data, gasLim)
 	// metrics should only be counted on a successful PFD tx
-	if response.Code == 0 && err == nil {
+	if response != nil && response.Code == 0 && err == nil {
```

Comment:
Bruh
This is the best DevX I've ever seen 😂

Reply:
its so bad... We should probably hunt down where they do this and open a PR, no?

Reply:
Wait.. we're just stupid. Let me fix

Reply:
😂